### PR TITLE
Add Harbor Docs

### PR DESCRIPTION
**Pull request by cdchris12**
This PR adds documentation on setting up and using Harbor within Lagoon as well as how security scanning works within Harbor.

# Checklist
- [] Documentation has been written/updated
- [X] PR title is ready for changelog and subsystem label(s) applied

### CODE REVIEW

File: docs/using_lagoon/harbor/harbor-container-settings/harbor-core.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
+# Harbor-Core Settings
```
Comment (AlannaBurke) [inline docs/using_lagoon/harbor/harbor-container-settings/harbor-core.md:1]: Are there any other setup instructions needed to get Harbor running, or will it just detect the existence of these files?

Comment (cdchris12) [reply docs/using_lagoon/harbor/harbor-container-settings/harbor-core.md:1]: The configuration is entirely handled by the config file and environment variables. Once those are in place, you'll have a fully functional local dev Harbor environment. In production, we have scripts which generate the ones which require a password and some logic to inject those secrets as required to make Harbor run as expected.

File: docs/using_lagoon/harbor/harbor-container-settings/harbor-core.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+  * This values sets how many days each issues robot token should be valid for.
+  * The default value is set to `999`.
+* `SYNC_REGISTRY`
+  * This value is not used.
+  * The default value is `false`.
+* `TOKEN_SERVICE_URL`
+  * The URL that the harbor-core service publishes to other services in order to retrieve a JWT token.
+  * The default value is `http://harbor-core:8080/service/token`.
+* `WITH_CHARTMUSEUM`
+  * Tells harbor-core if the Chartmuseum service is being used.
```
[the remark this answers is not in the document]
Comment (cdchris12) [reply docs/using_lagoon/harbor/harbor-container-settings/harbor-core.md:140]: Changed the other instance of it to match

File: docs/using_lagoon/harbor/harbor-container-settings/harbor-core.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+  * This service is **not** used with Lagoon's implementation of Harbor.
+  * The default value is `false`.
+* `WITH_CLAIR`
+  * Tells harbor-core if the harbor-clair service is being used.
```
[the remark this answers is not in the document]
Comment (cdchris12) [reply docs/using_lagoon/harbor/harbor-container-settings/harbor-core.md]: `harborclair` is correct

File: docs/using_lagoon/harbor/harbor_settings.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+* `HARBOR_REGISTRY_STORAGE_AMAZON_BUCKET`
+  * This needs to be set to the name of the AWS bucket which Harbor will save images to.
+  * Defaults to `harbor-images` when Lagoon is run locally or during CI testing.
+* `HARBOR_REGISTRY_STORAGE_AMAZON_REGION`
+  * This needs to be set to the AWS region in which Harbor's bucket is located.
+  * Defaults to `us-east-1` when Lagoon is run locally or during CI testing.
+* `REGISTRY_STORAGE_S3_ACCESSKEY`
+  * This needs to be set to the AWS access key Harbor should use to read and write to the AWS bucket.
+  * Defaults to an empty string when Lagoon is run locally or during CI testing, as MinIO does not require authentication.
```
[the remark this answers is not in the document]
Comment (cdchris12) [reply docs/using_lagoon/harbor/harbor_settings.md:18]: Resolved in https://github.com/amazeeio/lagoon/pull/1680/commits/04cff0c6fbc3d7a9b65432946a8e04c36f5171e4

File: docs/using_lagoon/harbor/harbor-container-settings/harbor-core.md (hunk 1 above)
Comment (AlannaBurke) [reply docs/using_lagoon/harbor/harbor-container-settings/harbor-core.md:1]: I think it would be good to put this ^ in the docs!

**Review by AlannaBurke — APPROVED** [review]

**Pull request merged**